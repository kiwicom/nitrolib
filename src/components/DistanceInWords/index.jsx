// @flow strict
import * as React from "react";
import formatDistance from "date-fns/formatDistance";
import * as locales from "date-fns/locale";

import { Consumer } from "../../services/intl/context";

type Props = {|
  time: Date | string,
|};

const DistanceInWords = (props: Props) => (
  <Consumer>
    {intl =>
      formatDistance(Date.now(), props.time, {
        locale: locales[intl.language.locations] || locales.enUS,
      })
    }
  </Consumer>
);

export default DistanceInWords;
