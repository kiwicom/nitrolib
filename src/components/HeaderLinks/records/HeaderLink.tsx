export type HeaderLink = {
  id: string,
  translation: string,
  newWindow: boolean,
  url: string,
};

export type SearchForm = {
  mode: string,
  destination: { type: string, name: string }, // TODO move proper search form into records @vacuum
  checkIn: Date,
  checkOut: Date | null,
  adults: number,
  children: number,
};

export type HeaderLinksContext =
  | "holidays"
  | "booking"
  | "kiwicom"
  | "pulkovoairport"
  | "kayakholidays";
