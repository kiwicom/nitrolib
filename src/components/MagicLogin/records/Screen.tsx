
export type Screen =
  | "intro"
  | "signUp"
  | "noAccount"
  | "kiwiLogin"
  | "facebookLogin"
  | "googleLogin"
  | "magicLink"
  | "signUpConfirmation"
  | "resetPassword"
  | "getSingleBooking";
