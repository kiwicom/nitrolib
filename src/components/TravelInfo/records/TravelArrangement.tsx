
export type Sector = {
  from: string,
  to: string,
  stops: "0" | "1" | "2" | "3",
  note: string,
};

export type TravelArrangement = {
  takeOff: Sector,
  landing: Sector,
};
