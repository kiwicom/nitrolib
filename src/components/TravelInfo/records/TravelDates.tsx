
export type TravelDates = {
  from: string,
  to: string,
};
