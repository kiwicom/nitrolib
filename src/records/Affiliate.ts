export type Affiliate = {
  id: string,
  params: { [key: string]: string },
};
