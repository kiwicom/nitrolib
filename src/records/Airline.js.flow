// @flow strict
export type Airline = {|
  id: string,
  lcc: number,
  name: string,
|};

export type Airlines = {|
  [key: string]: Airline,
|};

declare export var airlineDefault: Airline;
