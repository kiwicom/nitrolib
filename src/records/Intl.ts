import enUS from "date-fns/locale/en-US"; // fallback

import { langInfoDefault } from "./LangInfo";
import { LangInfo } from "./LangInfo";
import { Translations, Translate } from "../services/intl/translate";

export type IntlRaw = {
  language: LangInfo,
  translations: Translations,
};

export type Intl = {
  ...IntlRaw,
    translate: Translate,
    getLocale: Promise<$FlowFixMe>,
};

export type IntlsRaw = { [key: string]: IntlRaw };
export type Intls = { [key: string]: Intl };

// eslint-disable-next-line import/prefer-default-export
export const intlDefault: Intl = {
  language: langInfoDefault,
  translations: {},
  translate: id => id,
  getLocale: Promise.resolve(enUS),
};
