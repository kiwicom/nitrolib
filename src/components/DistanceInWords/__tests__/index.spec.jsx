// @flow strict
import * as React from "react";
import { shallow } from "enzyme";
import addHours from "date-fns/addHours";

import DistanceInWords from "..";

import { intlDefault } from "../../../records/Intl";

describe("#DistanceInWords", () => {
  test("#formatDistance", () => {
    const time = addHours(Date.now(), 10);
    const wrapper = shallow(<DistanceInWords to={time} />);

    expect(wrapper.prop("children")(intlDefault)).toMatchSnapshot();
  });
});
