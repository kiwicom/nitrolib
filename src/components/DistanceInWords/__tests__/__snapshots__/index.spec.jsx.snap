// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#DistanceInWords #formatDistance 1`] = `"about 10 hours"`;
